src/corecel/Assert.hh

```diff
@@ -237,7 +237,7 @@
             cudaError_t cuda_result_ = (STATEMENT);                \
             if (CELER_UNLIKELY(cuda_result_ != cudaSuccess))       \
             {                                                      \
-                cudaGetLastError();                                \
+                cuda_result_ = cudaGetLastError();                 \
```

Comment:
It feels weird to drop/ignore the return value of the function entirely, are we guaranteed that it is completely redundant with the cuda last error?

Reply:
Yeah it does to me too... but we have to call this function to clear the last error (so subsequent calls like in different tests don't fail). If something else (maybe asynchronous?) has failed in the meantime, then there's not really a good way of reporting that two different errors happened. And honestly given the number of times I've seen mismatched errors come out of the CUDA failures (e.g. a thrust error that seems to come from an unrelated celeritas kernel) it's probably not worth worrying about.

Reply:
I see.  I would still consider including the numeric value of the return value in the error message (for completeness)

Reply:
Is that suggestion unrelated to the "possibly two different errors" discussion above? Since we translate the enum into a descriptive string using the CUDA API, I also don't see how the numeric value could be helpful...

Reply:
> Is that suggestion unrelated to the "possibly two different errors" discussion

Yes.  I meant something like:
```
std::string cudaError( cudaGetErrorString(cudaGetLastError()) );
cudaError+= "(actual return code: " );
cudaError+= std::to_string(cuda_result_);
cudaError+= ");"
```
> Since we translate the enum into a descriptive string

If I remember correctly it often still include the numerical error code.

Reply:
If you think it's important enough, could you open a new PR with an implementation of your desired change showing the error messages before and after? Let's keep this discussion focused on the potential difference between the original function call return value and the return from `cudaGetLastError`.

Reply:
Fair enough.